zk/watcher.go

```diff
@@ -210,6 +210,18 @@ Reconnect:
 	w.cancelWatches()
 }
 
+func (w *Watcher) CreateIDAssignmentLock() *zk.Lock {
+	node := path.Join(w.prefix, "id_assignment_lock")
+
+	w.lock.RLock()
+	defer w.lock.RUnlock()
+
+	w.hooksLock.Lock()
```

Comment:
I don't think you need the `hooksLock`; that only protects `ephemeralNodes` and `watchedNodes`.